openfecli/parameters/plan_network_options.py

```diff
@@ -0,0 +1,179 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+"""Pydantic models for the definition of advanced CLI options
+
+"""
+import click
+from collections import namedtuple
+try:
+    # todo; once we're fully v2, we can use ConfigDict not nested class
+    from pydantic.v1 import BaseModel  # , ConfigDict
+except ImportError:
+    from pydantic import BaseModel
+from plugcli.params import Option
+from typing import Any, Optional
+import yaml
+import warnings
+
+
+PlanNetworkOptions = namedtuple('PlanNetworkOptions',
+                                ['mapper', 'scorer',
+                                 'ligand_network_planner', 'solvent'])
+
+
+class MapperSelection(BaseModel):
+    # model_config = ConfigDict(extra='allow', str_to_lower=True)
+    class Config:
+        extra = 'allow'
+        anystr_lower = True
+
+    method: Optional[str] = None
```

Comment:
Rather than validating later on, could we just use a validator here for the allowed method names?

Reply:
discussed this live, it won't be possible to have a list of possible methods a priori